lib/arrays.md

```diff
@@ -4,45 +4,53 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 ---
 
 # What is an Array?
+An Array is a Ruby data type that stores an ordered list of values.  
+An Array is a number-indexed list.  
+Values in a Array can be any type of object, including other Arrays.
 
-Replace me with your response...
 
 # What are some examples of information that would be Arrays as opposed to some other data type?
+Some examples of information that would be Arrays are a collection of values or list of data.
 
-Replace me with your response...
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
+We can retrieve the 6th element in an Array by using the #[5] method, the 8th element using the #[7] method.
+If we try to retrieve the value of an element that doesn't exist in our Array we will get => nil as a result.
 
-Replace me with your response...
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
+Assuming the Array contains at least 6 elements, we can find the -6th element.  The -6th element would be the 6th to last.
 
-Replace me with your response...
 
 # How would you perform an operation on every element inside an Array?
+Here we can perform an operation on every element inside an Array by using the Enumeral method '.map'.
+For example, array = [1,2,3,4,5,"Fiona"].map{ |element| element * 3} => [3, 6, 9, 12, 15, "FionaFionaFiona"]
```

Comment:
That's a valid way of looking at it. One thing you'll learn about later in the course is that when we program something, we want our commands to take the least effort possible in achieving their goal. If we compare what `each` and `map` do, then we might arrive at the following "script" for what happens internally when those commands are executed:

For `each`:

1. With the first element in the array...
2. Run some operation on that element.
3. Move on to the next element in the array.
4. Rinse and repeat :)
5. Return the original array.

For `map`:

1. Initialize a new array into which we'll store the values from each operation as we go through the original array's values.
2. With the first element in the original array...
3. Run some operation on that element.
4. Add the resulting value of the operation to the new array we made in Step 1.
5. Move on to the next element in the array.
6. Rinse and repeat :)
7. Return the new array from Step 1 (which by now has been populated with the results of each operation).

There is no revision to your answer needed, but take a moment to consider this difference between `each` and `map`. In this case, we would prefer `each` _unless_ we had a specific need to have the new array returned at the end. Otherwise the extra work needed to accomplish `map` is not worth it.